### DISCUSSION TITLE
Get access to the query/mutation parameters inside mapped type

### DISCUSSION THREAD

**Discussion by smilesrg** [thecodingmachine/graphqlite#522] [Q&A]
Hello!

How can query/mutation parameters be accessed inside a mapped type?

I'll give some real examples. There's a query:

```
query {
  feedback(classId:"41reqda11") {
    questions {
      answers {
        id
        answer
      }
      studentAnswers {
        id
        answer
      }
    }
  }
}
```

and the mapped type:
```php
/**
 * @Type(class=Question::class)
 * @SourceField(name="answers")
 */
class QuestionType
{
    /**
     * @return Answer[]
     */
    #[Field, Logged, Right(User::ROLE_STUDENT)]
    public function getStudentAnswers(
        Question $question
    ): array {
        // I need an access to the query parameter "classId" right here
    }
}
```
So I need to get all answers answered by the student for the given class. To do this, I need to know the classId. How can I fetch classId parameter?

**Comment by oojacoboo** [comment]
@smilesrg I would just write a db query that filters my questions by the `classId`.  Is that not possible?  That would be handled in your controller/repository.

**Reply by smilesrg** [reply]
Complicated: feedback has questions, and questions have answers. We need to find feedback answers given by the logged-in student for the given class id.

Feecback ---> Questions ---> Answers

**Reply by smilesrg** [reply]
As a workaround, I can use DTO classes for this, but why do this workaround? It really takes time

**Reply by oojacoboo** [reply]
I'm not sure I see the issue with a repository query.  Why can't you just do an `INNER JOIN` and get the correct results?

**Reply by smilesrg** [reply]
To query student answers, I need to get access to the `classId` value

**Reply by oojacoboo** [reply]
Why can't you use something like this?

```sql
SELECT DISTINCT q.*
FROM questions q
INNER JOIN answers a
    ON a.question_id = q.id
    AND a.class_id = :class_id
```

**Reply by smilesrg** [reply]
I need a class id, which I can't access in the `QuestionType`. It's what this whole conversation is about. See my code, I wrote right there:

```php
// I need an access to the query parameter "classId" right here
```

**Reply by oojacoboo** [reply]
I understand what you're asking.  But, I'm not convinced you're asking the right questions.  What do you plan to do with that `classId` within that model method?  Maybe prefetching will help you?  https://graphqlite.thecodingmachine.io/docs/prefetch-method

**Reply by smilesrg** [reply]
Pretty close, but then I have to modify the query slightly. I need to duplicate `classId` argument to the `studentAnswers`. however, duplicating the parameter is not a good idea.

```
query {
  feedback(classId:"41reqda11") {
    questions {
      answers {
        id
        answer
      }
      studentAnswers(classId:"41reqda11") {
        id
        answer
      }
    }
  }
}
```

**Reply by oojacoboo** [reply]
You'll have to outline your db schema/model more for me to be able to offer anymore insights here.  It sounds like you're trying to shoehorn in an implementation on top of a bad schema and avoiding using proper repository queries.  Maybe I'm wrong, but I'd need to see the schema and more details on how you're planning to expose it over the API.
